src/client.ts

```diff
@@ -600,4 +605,86 @@ export default class Client implements ClientInterface {
       onSave: pullOn === "save" || pullOn === "both",
     };
   }
+
+  private async benchmarkMiddleware<T>(
+    type: string | MessageSignature,
+    params: any,
+    runRequest: () => Promise<T>,
+  ): Promise<T> {
+    // Because of the custom bundle logic in the server, we can only fetch the server version after launching it. That
+    // means some requests may be received before the computed the version. For those, we cannot send telemetry
+    if (this.serverVersion === undefined) {
+      return runRequest();
+    }
+
+    const telemetryData: RequestEvent = {
+      request: typeof type === "string" ? type : type.method,
+      rubyVersion: this.ruby.rubyVersion!,
+      yjitEnabled: this.ruby.yjitEnabled!,
+      lspVersion: this.serverVersion,
+      requestTime: 0,
+    };
+
+    // If there are parameters in the request, include those
+    if (params) {
+      const castParam = { ...params } as { textDocument?: { uri: string } };
+
+      if ("textDocument" in castParam) {
+        const uri = castParam.textDocument?.uri.replace(
+          // eslint-disable-next-line no-process-env
+          process.env.HOME!,
+          "~",
+        );
+
+        delete castParam.textDocument;
+        telemetryData.uri = uri;
+      }
+
+      telemetryData.params = JSON.stringify(params);
+    }
+
+    let result: T | undefined;
+    let errorResult;
+
+    const benchmarkId = Math.floor(Math.random() * 1000);
```

Comment:
Is this 'random enough'? Seems we'd often have duplicates.

Reply:
This ID is just for marking the start and end of the benchmark. I switched to use an incrementing ID instead to avoid having to do the calculations.